packages/frontend/src/page/components/answer-form/answer/index.tsx

```diff
@@ -1,91 +1,158 @@
-import { NamespacedTranslateFunction } from "@carrot-kpi/react";
-import { Typography } from "@carrot-kpi/ui";
-import { BigNumber } from "ethers";
+import {
+    NamespacedTranslateFunction,
+    useNativeCurrency,
+} from "@carrot-kpi/react";
+import { Skeleton, Typography } from "@carrot-kpi/ui";
+import { BigNumber, utils } from "ethers";
 import { formatUnits } from "ethers/lib/utils.js";
 import { ReactElement } from "react";
 import { BYTES32_ZERO } from "../../../../commons";
 import {
     isAnsweredTooSoon,
     isAnswerInvalid,
-    isAnswerMissing,
     isAnswerPurelyBoolean,
     isQuestionFinalized,
     isAnswerPurelyNumerical,
+    isAnswerMissing,
 } from "../../../../utils";
 import { RealityQuestion } from "../../../types";
+import { AnswerInfo } from "../../answer-info";
 import { LearnMore } from "../../learn-more";
 
 interface AnswerProps {
     t: NamespacedTranslateFunction;
     question: RealityQuestion;
+    loadingQuestion: boolean;
 }
 
-export const Answer = ({ t, question }: AnswerProps): ReactElement => {
-    if (isAnswerMissing(question))
-        return (
-            <Typography variant="lg">{t("label.answer.missing")}</Typography>
-        );
-
+export const Answer = ({
+    t,
+    question,
+    loadingQuestion,
+}: AnswerProps): ReactElement => {
+    const nativeCurrency = useNativeCurrency();
     const finalized = isQuestionFinalized(question);
-    // TODO: display the bond somewhere
-    // const formattedBond = formatUnits(question.bond, nativeCurrency.decimals);
+
+    const formattedBond = formatUnits(question.bond, nativeCurrency.decimals);
+    const purelyBoolean = isAnswerPurelyBoolean(question);
+    const purelyNumerical = isAnswerPurelyNumerical(question);
+    const invalid = isAnswerInvalid(question);
+    const answeredTooSoon = isAnsweredTooSoon(question);
+
     return (
```

Comment:
Not really required, I wouldn't exactly do this now, but if we have a sort of internal subcomponent that handles various states using if/elses, it could be a bit more performant